src/network/core.rs

```diff
@@ -234,6 +234,173 @@ impl Core {
         }
     }
 
+    pub fn macvlan_per_podman_network(
+        per_network_opts: &types::PerNetworkOptions,
+        network: &types::Network,
+        netns: &str,
+    ) -> Result<types::StatusBlock, std::io::Error> {
+        //  StatusBlock response
+        let mut response = types::StatusBlock {
+            dns_search_domains: Some(Vec::new()),
+            dns_server_ips: Some(Vec::new()),
+            interfaces: Some(HashMap::new()),
+        };
+        // Does config have a macvlan mode ? I think not
+        // Important !! Hardcode MACVLAN_MODE to bridge
+        let macvlan_mode: u32 = 4u32;
+        // get master interface name
+        let master_ifname: String = network.network_interface.as_ref().unwrap().to_owned();
```

Comment:
I forgot to mention, network_interface is optional because we do not require it for macvlan. If it is empty you should use the default route interface. Here is the cni code https://github.com/containernetworking/plugins/blob/f1f128e3c9220634d3f26b96950271f87c34e424/plugins/main/macvlan/macvlan.go#L63

Reply:
@Luap99 thanks added a change in new commit.